### DISCUSSION TITLE
Why is this react-aria-components tanstack/table demo slow

### DISCUSSION THREAD

**Discussion by rostero1** [TanStack/table#4954] [General]
Devon put together a nice example of combining TanStack Table with react-aria-components.  Without the tanstack integration, the arrow keys are very fast.  They seem very slow with the integration and I can't figure out why.

https://codesandbox.io/p/sandbox/blissful-dubinsky-upkz00?file=%2Fsrc%2FApp.tsx

**Comment by greg-hoarau** [comment]
Hi, 

I've also noticed that the arrow keys are slow when using TanStack Table with React Aria Components. I've tried updating all dependencies during my test, but the issue persists. Would love to know if anyone knows what's happening. I can provide an updated version if necessary. 

It'd also be great to have a guide on this integration in the docs.

Thanks 🙏

**Comment by iamrakhmatov** [comment]
Hey, did anyone found the reason?

**Comment by dnaploszek** [comment]
I think it's due to the fact that react-aria uses javascript to store focus and hover states. If you inspect rerenders, the whole table rerenders as you use arrow keys. On hover it "just" rerenders the hovered cells even though in most cases this is just useless. We had some issues with the react-aria table and decided to move out until we notice that there're performance improvements in place. However it might not be as easy as it sounds.

**Reply by iamrakhmatov** [reply]
@devongovett, any improvements on this front?

**Comment by iamrakhmatov** [comment]
@devongovett do you guys have any solution (even temporary until you guys maybe fix it) to mitigate this performance issue?
